src/traces/box/attributes.js

```diff
@@ -203,5 +203,16 @@ module.exports = {
         },
         editType: 'plot'
     },
-    fillcolor: scatterAttrs.fillcolor
+    fillcolor: scatterAttrs.fillcolor,
+    hoveron: {
+        valType: 'flaglist',
+        flags: ['boxes', 'points'],
+        dflt: 'boxes',
```

Comment:
Debatable choice here. Strictly speaking we need `dflt: 'boxes'` to keep backward compat, but perhaps someone can argue otherwise.

Reply:
`dflt: 'boxes'` 👍

Reply:
I agree here, but perhaps to make this feature accessible to existing graphs, the _show closest data on hover_ mode bar button could restyle `hoveron` to `'points+boxes'` as well as `layout.hovermode` on click for traces with `boxpoints: 'all'`?

Reply:
oh, hmm... that's a good point actually. Can I change my answer to `dflt: 'boxes+points'`? I don't think turning on a new hover feature really amounts to a breaking change, I suppose in principle it can lead to events that a user application doesn't know how to handle but that seems pretty minor IMHO.

Reply:
`dflt: boxes+points` 👍  One could even argue that not being able to hover on the box points is a 🐞 .

Reply:
done inhttps://github.com/plotly/plotly.js/pull/2094/commits/6bfcbe0e44a36eb75548f259cb35a808c5671444